### PR TITLE
some const and inheritance correctness

### PR DESCRIPTION
**Pull request by 12xx12**
yeah - me IDE flashed this at me and I fixed it

i thought about enforcing more const correcness in more classes - i don't know if it's neccesary

### CODE REVIEW

File: src/Blocks/WorldInterface.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ class cWorldInterface
 	virtual void SendBlockTo(int a_BlockX, int a_BlockY, int a_BlockZ, cPlayer & a_Player) = 0;
 
 	/** Sends the block on those coords to the player */
-	inline void SendBlockTo(const Vector3i a_BlockPos, cPlayer & a_Player)
+	virtual inline void SendBlockTo(const Vector3i a_BlockPos, cPlayer & a_Player)
```
Comment (peterbell10) [inline src/Blocks/WorldInterface.h:64]: Nah, only one overload needs to be virtual.

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: src/Blocks/WorldInterface.h (hunk 1 above)
Comment (12xx12) [reply src/Blocks/WorldInterface.h:64]: Waht do you meen? Whe I remove the virtual expression we're back to square one and can't explicitly overide the function

Comment (tigerw) [reply src/Blocks/WorldInterface.h:64]: In fact I think you can delete the body of this altogether and make it pure virtual

Comment (peterbell10) [reply src/Blocks/WorldInterface.h:64]: `SendBlockTo(int, int, int)` can and should do the exact same thing as `SendBlockTo(Vector3i)`, so there's no reason for this to be virtual.

Comment (12xx12) [reply src/Blocks/WorldInterface.h:64]: But the method in cWorld shadows this method in cWorldInterface. This was the reason for this PR

**Comment by tigerw** [comment]
peterbell’s point perhaps is that we should delete the overridden method in cWorld and keep the implementation in cWorldInterface, not marked as virtual, acting purely as a forwarder?

Sent from my Windows 10 Mobile ??

**Review by tigerw — APPROVED** [review]

**Pull request merged**